lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -31,24 +30,21 @@
 {
     private final CompressionCodecName codec;
     private final long valueCount;
-    private final List<DataPage> compressedPages;
+    private final LinkedList<DataPage> compressedPages;
```

Comment:
wouldn't ArrayDeque be better

Reply:
Probably not in this case.

Since each DataPage removed is processed completely before the next access to the list- I would expect the CPU cache to be mostly thrashed by that processing between accesses and preventing the memory locality of array based collections from helping (plus the array allocation can’t shrink as items are removed).

Reply:
Would that be the second case in my life where LinkedList is actually useful? :) 

> preventing the memory locality of array based collections

i was thinking mostly about the overhead of the linkedlist for add one, remove one.

whatever.